src/routes/authentication/components/FormWrapper.component.tsx

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useState } from "react";
+import {
+  Box,
+  Avatar,
+  Container,
+  Typography,
+  Grid,
+  Button,
+} from "@mui/material";
+import LockOutlinedIcon from "@mui/icons-material/LockOutlined";
+import { Link } from "react-router-dom";
+import styled from "styled-components";
+import SignUpFormFields from "./AuthFormFields";
+import Verification from "./Verification.component";
+
+const SIGN_UP_FORM_FIELDS = [
+  { name: "firstName", label: "First Name", type: "text", sm: 6, xs: 12 },
+  { name: "lastName", label: "Last Name", type: "text", sm: 6, xs: 12 },
+  { name: "email", label: "Email", type: "email", xs: 12 },
+  { name: "postCode", label: "Post Code", type: "text", xs: 12 },
+  { name: "password", label: "Password", type: "password", xs: 12 },
+];
+
+const SIGN_IN_FORM_FIELDS = [
+  { name: "email", label: "Email", type: "email", xs: 12 },
+  { name: "password", label: "Password", type: "password", xs: 12 },
+];
+
+type FormWrapperProps = {
+  handleFormFieldChange: (
+    e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
+  ) => void;
+  authType: string;
+};
+
+const StyledBox = styled(Box)`
+  margin-top: var(--spacing-xs);
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const StyledAvatarWrapper = styled(Avatar)`
+  background-color: var(--color-deep-purple);
+  margin-bottom: var(--spacing-sm);
+`;
+
+const StyledLink = styled(Link)`
+  color: var(--color-black);
+`;
+
+const StyledButton = styled(Button)`
+  margin-bottom: var(--spacing-md);
+`;
+
+const FormWrapper = (props: FormWrapperProps) => {
+  const { handleFormFieldChange, authType } = props;
+  const [showVerification, setShowVerification] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (authType === "sign-up") {
+      setShowVerification(true);
```

Comment:
I kind of agree with you on not loving how verification is displayed, how come you didn't make it a seperate page with a seperate route? I don't hate it though, it seems to work well which is the main thing. Maybe even rename this file to AuthFormWrapper just to make it clear it's not a generic reusable component

Reply:
To keep FormWrapper reusable you could pass it a prop called onSubmit. Then on /signUp we'd have `<FormWrapper onSubmit={() => navigate("/validate")} />`.

We could also add props for formFields, title, submitText etc and then in Authentication have (not real Javascript)
```
switch(authType) {
    case AuthType.SignIn: <FormWrapper formFields={SIGN_IN_FORM_FIELDS} title={"Sign in"} ... />
    case AuthType.SignUp: ....
}
```